activerecord/lib/active_record/associations/collection_proxy.rb

```diff
@@ -1110,6 +1111,7 @@ def reload
       def reset
         proxy_association.reset
         proxy_association.reset_scope
+        @offsets = {}
         reset_scope
```

Comment:
Originally I intended to clear `@offsets` in `reset_scope`. https://github.com/rails/rails/pull/29098#discussion_r118800958
Do you intend to keep `@offsets` in `reset_scope`?